File: codeforces/cf1703C_cypher.cpp
// Vicfred
// https://codeforces.com/contest/1703/problem/C
// implementation
#include <iostream>
#include <vector>

using namespace std;

int main() {
  int t;
  cin >> t;
  while (t--) {
    int n;
    cin >> n;
    vector<int> a(n);
    for (int i = 0; i < n; ++i) {
      cin >> a[i];
    }
    for (int i = 0; i < n; ++i) {
      int b;
      cin >> b;
      string s;
      cin >> s;
      for (const auto &ch : s) {
        if (ch == 'D') {
          a[i]++;
        } else {
          a[i]--;
        }
      }
      cout << (a[i] + 10) % 10 << " ";
    }
    cout << endl;
  }
  return 0;
}

File: codeforces/cf1703C_cypher.rb
# uninhm
# https://codeforces.com/contest/1703/problem/C
# implementation

t = gets.to_i

t.times do
  n = gets.to_i
  a = gets.split.map &:to_i
  n.times do |i|
    b = gets.chomp
    a[i] = (a[i] + b.count('D') - b.count('U')) % 10
  end
  puts a*' '
end
